src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs

```diff
@@ -81,8 +80,16 @@ void SetReplyToAddress(OutgoingMessage outgoingMessage, BrokeredMessage brokered
 
                     if (selected != null)
                     {
-                        replyTo = mapper.Map(new EntityAddress(replyTo.Name, selected.Name));
+                        if (mapper is DefaultNamespaceNameToConnectionStringMapper)
```

Comment:
> var convertedValue = mapper.Map(replyTo); does not translate name into connectionstring, it does the inverse and blow up if the name is not found, that is not what is needed for backward compat.

It isn't true. `Mapper` injected into `DefaultBatchedOperationsToBrokeredMessagesConverter` could be: 
- `DefaultNamespaceNameToConnectionStringMapper` if `UseNamespaceNameInsteadOfConnectionString` is off. This mapper converts back `namespace name` to `connection string`
- `PassThroughNamespaceNameToConnectionStringMapper`  if `UseNamespaceNameInsteadOfConnectionString` is enabled and it doesn't do anything on provided value (that should be a namespace name)

> It's also problematic if the suffix is a connectionstring from the beginning (if user does .Send("ep@fullconnectionstring") and not a name), in that case no mapping is registered and it would also blow up.

I didn't think about this scenario, but in that case I suspect that [this line](https://github.com/Particular/NServiceBus.AzureServiceBus/blob/2e577f3e44f96bf5a8db8872b7b7f71ae3fc2a2c/src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs#L69) will throw

Anyway...how do we want to behave in this "mixed scenario": user enables `UseNamespaceNameInsteadOfConnectionString` to improve security and then they'll use `.Send("ep@fullConnectionString")` without configuring namespace name?
It's inconsistent. I'd prefer to throw rather than send it out